spec/mem/tree_borrows/memory.md

```diff
@@ -90,7 +90,8 @@ impl<T: Target> TreeBorrowsMemory<T> {
         frame_extra: &mut TreeBorrowsFrameExtra,
     ) -> Result<Pointer<TreeBorrowsProvenance>> {
         let Some((alloc_id, parent_path)) = ptr.provenance else {
-            return ret(ptr);
+            if pointee_size.is_zero() { return ret(ptr);}
```

Comment:
```suggestion
            // Pointers without provenance cannot access any memory, so giving them a new
            // tag makes no sense. If the pointee also has size zero, this is fine, otherwise UB.
            if pointee_size.is_zero() { return ret(ptr);}
```